File: app/src/main/java/me/gilo/starter/common/CompletionDocLiveData.java
package me.gilo.starter.common;


import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentReference;


public final class CompletionDocLiveData extends LiveData<Resource<Boolean>> implements OnCompleteListener<DocumentReference> {


    public CompletionDocLiveData() {
        setValue(new Resource<>(Status.LOADING));
    }

    @Override
    public final void onComplete(@NonNull Task<DocumentReference> task) {
        if (task.isSuccessful()) {
            setValue(new Resource<>(true));
        } else {
            setValue(new Resource<>(task.getException()));
        }
    }
}

File: app/src/main/java/me/gilo/starter/common/CompletionDocLiveData.kt
package me.gilo.starter.common


import androidx.lifecycle.LiveData
import com.google.android.gms.tasks.OnCompleteListener
import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.DocumentReference


class CompletionDocLiveData : LiveData<Resource<Boolean>>(), OnCompleteListener<DocumentReference> {
    init {
        value = Resource(Status.LOADING)
    }

    override fun onComplete(task: Task<DocumentReference>) {
        if (task.isSuccessful) {
            value = Resource(true)
        } else {
            value = Resource(task.exception!!)
        }
    }
}
